snow/engine/snowman/bootstrap/bootstrapper.go

```diff
@@ -378,24 +372,20 @@ func (b *Bootstrapper) startSyncing(ctx context.Context, acceptedContainerIDs []
 	// Initialize the fetch from set to the currently preferred peers
 	b.fetchFrom = b.StartupTracker.PreferredPeers()
 
-	pendingContainerIDs := b.Blocked.MissingIDs()
-	// Append the list of accepted container IDs to pendingContainerIDs to ensure
-	// we iterate over every container that must be traversed.
-	pendingContainerIDs = append(pendingContainerIDs, acceptedContainerIDs...)
+	b.missingBlockIDs.Add(acceptedContainerIDs...)
+	numMissingBlockIDs := b.missingBlockIDs.Len()
 
 	log := b.Ctx.Log.Info
 	if b.restarted {
 		log = b.Ctx.Log.Debug
 	}
 	log("starting to fetch blocks",
 		zap.Int("numAcceptedBlocks", len(acceptedContainerIDs)),
-		zap.Int("numMissingBlocks", len(pendingContainerIDs)),
+		zap.Int("numMissingBlocks", numMissingBlockIDs),
 	)
 
-	toProcess := make([]snowman.Block, 0, len(pendingContainerIDs))
-	for _, blkID := range pendingContainerIDs {
-		b.Blocked.AddMissingID(blkID)
-
+	toProcess := make([]snowman.Block, 0, numMissingBlockIDs)
+	for blkID := range b.missingBlockIDs {
 		// TODO: if `GetBlock` returns an error other than
```

Comment:
q: given the rework, are we planning to solve this TODO as well? (maybe in a different PR)